how-it-works/7_smart_contracts/01_canister-lifecycle.subpage.md

```diff
@@ -0,0 +1,40 @@
+
+Charging for bandwidth is also done at the moment of use.  When a canister wants to send a request to another canister, the system calculates the number of cycles that sending the message costs (the cost of sending has a fixed component and a component that depends on the size of the payload) and deducts the cost from the canister’s balance.
+Furthermore, it also deducts the cost of sending a maximal size reply from the callee since for intercanister messages the caller pays for reply.  The cycles corresponding to the difference between the maximal size and the actual size of the reply are refunded to the canister when the reply arrives.
+
+When canisters run out of cycles, they are essentially deleted.  To avoid that deletion happens too suddenly, canisters have associated a so-called freezing threshold.  Once the canister’s balance dips below the freezing threshold (through charging by the system) then the canister stops processing any new requests; replies are still being processed. The system throws an error if at any time a canister attempts to perform an action (e.g. attaching cycles or sending a message) which would result in the cycles balance dipping below the freezing threshold.
+
+
+
+## Canister management
+
+Canisters are managed by controllers which can be users or even other canisters. Controllers are in charge of deploying and maintaining the canisters to the IC and they are the only entities who are allowed to perform management operations on canisters.
+The most common such operations are deploying a canister smart contract to the IC and  starting and stopping canisters. The controller of canisters can change the canister parameters, including adding and removing controllers, changing the freezing threshold.
+
+
+Controllers can update the code that runs on canisters by submitting a new Wasm module which should replace the older one.  By default, updating the Wasm module of a canister wipes out the heap but the content of the stable memory remains unchanged. The IC offers an upgrade mechanism where three actions are executed atomically: serializing the heap of the canister and writing it to stable memory, installing the new Wasm code and then deserializing the content of the stable memory.
+Of course, a canister may ensure at all times that the data that needs to be persisted across upgrades is stored in the stable memory in which case the upgrade process is significantly simpler. 
+
+A canister can have no controllers: such canisters are completely immutable and resemble traditional smart contracts.
```

Comment:
As in the abstract, I would probably highlight the different possibilities for controllers right after introducing the term.

Reply:
Done